src/services/populationService.js

```diff
@@ -1,39 +1,15 @@
 import { EventTargetShim } from '../utils/EventTargetShim.js';
 
 class PopulationService extends EventTargetShim {
-  constructor() {
-    super();
-    this.cache = {};
-  }
-
-  async getPopulation(opts) {
-    let { iso2 } = opts;
-    iso2 = encodeURI(iso2);
-
-    if (!/^[a-zA-Z]{2}$/.test(iso2)) {
-      return;
-    }
-
-    if (opts.forceRefresh || this.cache[iso2]?.status === 'failed' || !this.cache[iso2]) {
-      try {
-        this.cache[iso2] = {};
-        const res = await fetch(new URL('../../data/population.json', import.meta.url)).then(r => r.json());
-        this.cache[iso2] = {
-          status: 'success',
-          totalPopulation: res.data[iso2].Population ?? 0
-        };
-
-        return this.cache[iso2];
-      } catch (_) {
-        this.cache[iso2] = {
-          status: 'failed'
-        };
-      }
+  async getPopulation(forceRefresh) {
+    if (forceRefresh || !this.__population) {
+      this.__population = fetch(new URL('../../data/population.json', import.meta.url)).then(r => r.json());
```

Comment:
It would be nice to have some error handling and return a `status: 'failed'` if the request doesnt work. That way we can check for it here: https://github.com/Code-for-All/lockdown/blob/7f8e86f7e6f383c1ad9d75dea8449d692db4b937/src/components/CountryInfo.js#L96

Reply:
sure, will do